src/main/java/com/fasterxml/jackson/databind/DatabindContext.java

```diff
@@ -148,6 +148,16 @@ public abstract class DatabindContext
      */
     public abstract DatabindContext setAttribute(Object key, Object value);
 
+    /**
+     * Method to configure a provider for custom cache implementations downstream.
+     * Should be overriden by sub-classes. 
+     * 
+     * @since 2.16
+     */
+    public DatabindContext withCacheProvider(CacheProvider cacheProvider) {
```

Comment:
I don't think we want to attach `CacheProvider` to contexts, as long as it is available via `MapperConfig` -- for convenience we probably do want accessor, but that should only access it from underlying Config object.